### DISCUSSION TITLE
Requêtes par Thierry

### DISCUSSION THREAD

**Discussion by AlZorba** [esag-swiss/iDig-Webapp#1] [Ideas]
- [x] Champs: implémenter un système d'accordéon par groupe et non sur l'ensemble des champs.
- [x] Champs: préciser avec un alt qu'il s'agit des champs disponibles pour le type sélectionné.
- [x] Champs: corriger le bug des checkbox checked qui ne correspondent pas aux colonnes
- [x] Reset des champs à afficher dans le tableau (Trench>Identifiant>Titre) lors de changement de type. voir les possibilité de customisation utilisateur (cookies? fichier json? dans préférences?)
- [x] Colonne "menu" de gauche 'sticky' : fait avec div class="sticky-top"
- [] Seuls les champs du fichier préférences sont affichés, quid des autres ? Prévoir celui de UUID ? Le champ Attachments (aka RelationIncludesUUID) a déjà été ajouté en prévision de l'affichage des images.
- [x] Afficher par défaut les colonnes Trench>Identifiant>Titre, le Type est superflue.
- [] Ordre des colonnes modulable avec un drag & drop.
- [x] Largeur du tableau non contrainte. : fait avec div id="app" class="container-fluid"
- [x] Ajouter fenêtre de détails sur un clic.
- [x] Éventuellement tronquer les champs multilignes (l'intégralité peut être consultée par clic)
- [x] Aligner le texte des cellules en haut.
- [] Tester labels seulement (et pas label) pour l'affichage des champs
- [x] Déroulement indépendant fenêtres droite et gauche
- [] Colonnes : croix pour les virer, garder les colonnes en mémoire et les reseter, largeur reglable
- [] Pour afficher 15 colonnes, au lieu de toutes les squeezer dans la largeur de l'écran, il faudrait donner la possibilité de scroller vers la droite.
- [x] Implémenter recherche ciblée comme iDig, ce serait utile d'implémenter des critères de recherche (*%!) et la possibilité de filtrer par champ spécifique ("Description:terre cuite"). Et pourrais-tu ajouter un bouton "+" pour afficher des boîtes de requête supplémentaire qui permettent de filtrer un subset des données?
- [] Interdire l'édition sur IUUD
- [] Faire une "belles fiche" modulable
- [] Petit bug, la fenêtre est figée en haut de l'écran, donc quand tu scrolles en bas de ton tableau et que tu cliques sur une entrée, tu ne vois pas le popup
- [] Faire un tableau de comptage par contextes
- [] D'ailleurs, cette configuration par chaque utilisateur des champs qu'il souhaite afficher est top, mais sera-t-il aussi possible à l'avenir de sauvegarder une "configuration ESAG" que les utilisateurs pourront charger s'ils le souhaitent? 
- [x] "Trenches", trouver un moyen de regrouper les trenches par campagnes, comme dans iDig, avec le même système que implémenté pour les "Champs"
- [] Gérer port et http 
- [x] Récupérer les pref depuis idig-server
- [x] Formulaire par défaut avec champs indiqués dans préf (comme colonne de gauche)
- [] Formulaire ouvert dans nouvel onglet

**Comment by esag-admin** [comment]
-> pour la fonction de drag-&-drop les colonnes de la table: [https://www.davidecattani.dev/17-05-2021-how-to-add-drag-drop-functionality-to-bootstrapvue-datatable-columns/]

**Reply by AlZorba** [reply]
Ca à l'air sympa, je vais voir si ça match avec le composant vuejs utilisé "Tablelite".

**Comment by AlZorba** [comment]
Feedback of January 2024 :

- Feature : Spinner en barre de progression : barre de progression basée sur le nombre de trenches qui reste à télécharger
- Feature : sélection par années (□ AMA23, □ AMA22)  
- Bug ? : Le champs "relation" est toujours vide, quels que soient les types sélectionnés; est-ce parce qu'il s'agit d'un champ à valeur multiple qui n'est pas encore géré par la webapp?  il faut que je creuse...

**Comment by AlZorba** [comment]
Je met cette discussion dans "road map"

**Comment by AlZorba** [comment]
all done
